doc/tutorials/acrn-secure-boot-with-grub.rst

```diff
@@ -0,0 +1,256 @@
+    set timeout_style=menu
+    set timeout=5
+    set gfxmode=auto
+    set gfxpayload=keep
+    terminal_output gfxterm
+
+    menuentry "ACRN Multiboot Ubuntu Service VM" --users "" --id ubuntu-service-vm {
+
+    search --no-floppy --fs-uuid --set 3df12ea1-ef12-426b-be98-774665c7483a
+
+    echo 'loading ACRN...'
+    multiboot2 /boot/acrn/acrn.bin  root=PARTUUID="c8ee7d92-8935-4e86-9e12-05dbeb412ad6"
+    module2 /boot/bzImage Linux_bzImage
+    }
+
+Use the output of the :command:`blkid` to find the right values for the
+UUID (``--set``) and PARTUUID (``root=PARTUUID=`` parameter) of the root
+partition (e.g. `/dev/nvme0n1p2`) according to your your hardware.
+
+Copy this new :file:`grub.cfg` to your ESP (e.g. `/boot/efi/EFI/`).
+
+
+.. rst-class:: numbered-step
+
+Sign grub.cfg and ACRN Binaries
+*******************************
+
+The :file:`grub.cfg` and all ACRN binaries that will be loaded by GRUB
+**must** be signed with the same GPG key.
+
+Here's sequence example of signing the individual binaries::
+
+    gpg --homedir keys --detach-sign path/to/grub.cfg
+    gpg --homedir keys --detach-sign path/to/acrn.bin
+    gpg --homedir keys --detach-sign path/to/sos_kernel/bzImage
+
+Now, you can reboot and the system will boot with the signed GRUB EFI binary.
+GRUB will refuse to boot if any files it attempts to load have been tampered
+with.
+
+
+.. rst-class:: numbered-step
+
+Enable UEFI Secure Boot
+***********************
+
+Creating UEFI Secure Boot Key
+=============================
+
+-Generate your own keys for Secure Boot::
+
+    openssl req -new -x509 -newkey rsa:2048 -subj "/CN=PK/"  -keyout PK.key  -out PK.crt  -days 7300 -nodes -sha256
+    openssl req -new -x509 -newkey rsa:2048 -subj "/CN=KEK/" -keyout KEK.key -out KEK.crt -days 7300 -nodes -sha256
+    openssl req -new -x509 -newkey rsa:2048 -subj "/CN=db/"  -keyout db.key  -out db.crt  -days 7300 -nodes -sha256
+
+-Convert ``*.crt`` keys to the ESL format understood for UEFI::
+
+    cert-to-efi-sig-list PK.crt PK.esl
+    cert-to-efi-sig-list KEK.crt KEK.esl
+    cert-to-efi-sig-list db.crt db.esl
+
+-Sign ESL files::
+
+    sign-efi-sig-list -k PK.key -c PK.crt PK PK.esl PK.auth
+    sign-efi-sig-list -k PK.key -c PK.crt KEK KEK.esl KEK.auth
+    sign-efi-sig-list -k KEK.key -c KEK.crt db db.esl db.auth
+
+The keys to be enrolled in UEFI firmware: :file:`PK.der` , :file:`KEK.der` , :file:`db.der`.
+The keys to sign bootloader image :file:'grubx64.efi': :file:`db.key` , :file:`db.crt`.
+
+Sign GRUB Image With ``db`` Key
+================================
+
+    sbsign --key db.key --cert db.crt path/to/grubx64.efi
+
+:file:`grubx64.efi.signed` will be created, it will be your bootloader.
+
+Enroll UEFI Keys To UEFI Firmware
+=================================
+
+Enroll ``PK``(:file:`PK.der`), ``KEK``(:file:`KEK.der`) and ``db``(:file:`db.der`) in
+Secure Boot Configuration UI, which depends on your platform UEFI firmware. From now on,
```

Comment:
We should include an example of how to "Enroll" these files on one of our supported platforms, for example, the one you validated listed at the top of the page.

Reply:
@dbkinder  have updated this part, please review, thank you.
-Yonghua